tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -71,6 +71,34 @@ using gpuError_t = hipError_t;
 #endif
 
 namespace tensorflow {
+// Launches a GPU kernel through cudaLaunchKernel in Cuda environment, or
+// hipLaunchKernel in ROCm environment with the given arguments.
+//
+// The kernel parameters 'Ts' must be constructible from the arguments 'Args'.
+template <typename... Ts, typename... Args>
+Status GpuLaunchKernel(void (*function)(Ts...), dim3 grid_dim, dim3 block_dim,
+                        size_t shared_memory_size_bytes, gpuStream_t stream,
+                        Args... arguments) {
+  static_assert(detail::NoneIsReference<Ts...>(),
+                "Kernels with reference arguments have undefined behaviour.");
+  auto func_ptr = absl::bit_cast<const void*>(function);
+#if GOOGLE_CUDA
+  // Cast arguments and forward them as an array of pointers.
+  auto args_tuple = std::tuple<Ts...>(arguments...);
+  auto arg_ptrs = detail::GetArrayOfElementPointers(&args_tuple);
+  auto result = cudaLaunchKernel(func_ptr, grid_dim, block_dim, arg_ptrs.data(),
+                                 shared_memory_size_bytes, stream);
+  if (result != cudaSuccess) {
+    return errors::Internal(cudaGetErrorString(result));
+  }
+#elif TENSORFLOW_USE_ROCM
+  hipLaunchKernelGGL(func_ptr, grid_dim, block_dim, shared_memory_size_bytes,
+                     stream, std::forward<Args>(arguments)...);
+#endif
+  return Status::OK();
+}
+CREATE_CUDA_HOST_FUNCTION_ALIAS(GpuLaunchKernel, CudaLaunchKernel);
```

Comment:
I think for this to work, you will also need `cudaStream_t` type alias for `gpuStream_t`....do we have that?

Reply:
Right there is that from line 60 -71

Reply:
right you are!